### PR TITLE
Use PrintData to pass arguments to HTMLFormatter

### PR DESCRIPTION
**Pull request by tevincent**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/utils/HtmlFormatter.kt [hunk 1]
```diff
@@ -29,9 +29,8 @@ import org.jsoup.nodes.TextNode
 import com.google.android.material.R as RMaterial
 
 class HtmlFormatter(
-    private val context: Context? = null,
     private val html: String,
-    private val message: Message? = null,
+    private val printData: PrintData? = null
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/HtmlFormatter.kt:33]:
```suggestion
    private val printData: PrintData? = null,
```

File: app/src/main/java/com/infomaniak/mail/utils/WebViewUtils.kt [hunk 2]
```diff
@@ -50,23 +50,19 @@ class WebViewUtils(context: Context) {
     private val jsBridgeScript by lazy { context.getJsBridgeScript() }
 
     fun processHtmlForPrint(
-        context: Context? = null,
         html: String,
-        isDisplayedInDarkMode: Boolean,
-        message: Message? = null
-    ): String = with(HtmlFormatter(context = context, html = html, message = message)) {
-        addCommonDisplayContent(isDisplayedInDarkMode)
+        printData: HtmlFormatter.PrintData
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/WebViewUtils.kt:54]:
```suggestion
        printData: HtmlFormatter.PrintData,
```

File: app/src/main/java/com/infomaniak/mail/utils/WebViewUtils.kt [hunk 3]
```diff
@@ -50,23 +50,19 @@ class WebViewUtils(context: Context) {
...
+        ): String = with(HtmlFormatter(html = html, printData = printData)) {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/WebViewUtils.kt:55]: Format seems wrong here.

File: app/src/main/java/com/infomaniak/mail/utils/WebViewUtils.kt [hunk 4]
```diff
@@ -50,23 +50,19 @@ class WebViewUtils(context: Context) {
...
         return@with inject()
     }
 
     fun processHtmlForDisplay(
-        context: Context? = null,
         html: String,
         isDisplayedInDarkMode: Boolean,
-        message: Message? = null
-    ): String = with(HtmlFormatter(context = context, html = html, message = message)) {
+    ): String = with(HtmlFormatter(html = html)) {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/WebViewUtils.kt:65]:
```suggestion
    ): String = with(HtmlFormatter(html)) {
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/utils/WebViewUtils.kt [hunk 5]
```diff
@@ -50,23 +47,19 @@ class WebViewUtils(context: Context) {
...
     fun processHtmlForPrint(
-        context: Context? = null,
         html: String,
-        isDisplayedInDarkMode: Boolean,
-        message: Message? = null
-    ): String = with(HtmlFormatter(context = context, html = html, message = message)) {
-        addCommonDisplayContent(isDisplayedInDarkMode)
-        registerIsForPrint()
+        printData: HtmlFormatter.PrintData
+    ): String = with(HtmlFormatter(html = html)) {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/utils/WebViewUtils.kt]:
```suggestion
    ): String = with(HtmlFormatter(html)) {
```

**Review by LunarX — APPROVED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**